gwcs/lookup_table.py

```diff
@@ -0,0 +1,113 @@
+import numpy as np
+
+import astropy.units as u
+from astropy.modeling.core import Model, _prepare_inputs_single_model
+
+__all__ = ['LookupTable']
+
+
+class LookupTable(Model):
+    """
+    This model indexes the lookup table with the rounded value.
+    The primary difference between this model and ``Tabular`` models is that it
+    supports non-numeric values in the lookup table.
+
+    .. note::
+        Any units on the input value are ignored.
+
+    Parameters
+    ----------
+    lookup_table : `~astropy.units.Quantity` or `numpy.ndarray`
+
+    """
+
+    linear = False
+    fittable = False
+    standard_broadcasting = False
+
+    inputs = ('x',)
+    outputs = ('y',)
+
+    @property
+    def return_units(self):
+        if hasattr(self.lookup_table, 'unit'):
+            return {'y': self.lookup_table.unit}
+        else:
+            return None
+
+    def __init__(self, lookup_table):
+        super().__init__()
+
+        if not isinstance(lookup_table, u.Quantity):
+            lookup_table = np.asarray(lookup_table)
+
+        self.lookup_table = lookup_table
+
+    def evaluate(self, point):
+        ind = int(np.asarray(np.round(point)))
```

Comment:
Doesn't this prevent it from evaluating on array of inputs? It looks like this is on purpose but I wonder why.

I think it's good to document the chosen way of rounding the input, as there may be different ways.